snapcraft/tests/matchers.py

```diff
@@ -18,16 +18,35 @@
 import testtools
 
 
-class HasArchitecture:
-    """Match if the file was built for the expected architecture"""
-
-    def __init__(self, expected_arch):
-        self._expected_arch = expected_arch
+class HasMagic:
```

Comment:
Can you please explain this in a comment in the source code?
I'm totally confused because it's called magic.

Ideally, you could find a better name for this that's not the name of the module, which is pretty bad. Something like HasType? HasMIMEType?